src/pdf_parser/pdf_utils/disambiguate_layout.py

```diff
@@ -0,0 +1,372 @@
+from layoutparser import Layout, TextBlock, Rectangle, Detectron2LayoutModel
+from PIL.PpmImagePlugin import PpmImageFile
+from shapely.geometry import Polygon
+from shapely.ops import unary_union
+from typing import List, Tuple, Optional
+
+from pydantic import BaseModel as PydanticBaseModel
+
+
+class BaseModel(PydanticBaseModel):
+    """Base class for all models."""
+
+    class Config:
+        """Pydantic config."""
+
+        arbitrary_types_allowed = True
+
+
+# TODO: I added this because I want to enforce that the unexplained fractions are in the same order as the boxes in
+#  the layout without adding it as page metadata, as this would require formally writing checks
+#   with assert. Is there a better way to do this or is this ok?
+class LayoutWithFractions(BaseModel):
+    """Layout with unexplained fractions added."""
+
+    layout: Layout
+    unexplained_fractions: List[float]
```

Comment:
Thoughts on this? Wanted to create a different type than Layout for the "combine_layouts" function (see below). The alternative is to throw the fractions in page data attribute and then throwing assertion error at the start of a function if it doesn't have these attributes. What's a more canonical way of doing this?

Reply:
it seems from `combine_layouts` like you want to simultaneously loop through layouts and fractions. Worth using a dict to map layouts to fractions?